src/services/navigationBar.ts

```diff
@@ -188,11 +197,19 @@ namespace ts.NavigationBar {
                         return true;
                     }
 
-                    // Or if it is not parented by another function.  i.e all functions
-                    // at module scope are 'top level'.
+                    // Or if it is not parented by another function(except for parent functions that
+                    // are methods and constructors). I.e all functions at module scope are 'top level'.
                     if (!isFunctionBlock(functionDeclaration.parent)) {
                         return true;
                     }
+                    else {
+                        const grandParentKind = functionDeclaration.parent.parent.kind;
+                        if (grandParentKind === SyntaxKind.MethodDeclaration ||
+                            grandParentKind === SyntaxKind.Constructor) {

```

Comment:
My assumption is never, it covers them and more function like nodes. That's why in the else clause I need to filter out only method declarations and constructors.